File: 5/5.py
nice = 0

lines = [line.rstrip() for line in open('input')]

for line in lines:
    naughty = False

    if len(re.findall('(.*?[aeiou])', line)) < 3:
        continue

    if len(re.findall('(.)\1{1,}', line)) < 1:
        continue

File: 2015/5/5.rb
nice = 0

File.readlines('input').each do |line|
  next if line.scan(/(.*?[aeiou])/).length < 3
  next if line.scan(/(.)\1{1,}/).length < 1
  next if line.scan(/ab|cd|pq|xy/).length > 0

  nice += 1
end

puts nice
